napari/components/overlays/base.py

```diff
@@ -0,0 +1,20 @@
+from ...utils.events import EventedModel
+from .._viewer_constants import CanvasPosition
+
+
+class Overlay(EventedModel):
+    visible: bool = False
+    opacity: float = 1
+    order: int = 1e6
+
+    def __hash__(self):
+        return id(self)
+
+
+class CanvasOverlay(Overlay):
+    position: CanvasPosition = CanvasPosition.BOTTOM_RIGHT
+
+
+class SceneOverlay(Overlay):
+    # TODO: should transform live here?
```

Comment:
Can you give an example of a scene overlay transform? Maybe the interaction box transform?

Reply:
I'm not convinced of what should be our "default" here; for example, the `InteractionBox` (or `BoundingBoxOverlay` from #4849 which will probably be merged with it) do not need a transform that is directly settable by the user (despite *having* an affine themselves), since they derive it from the layer, and should never be offset manually. On the other hand, stuff like the manipulators from [`alisterburt/napari-threedee`](https://github.com/alisterburt/napari-threedee) should have an easily settable transform. It's probably something the subclass should expose, though.

Reply:
Given we don't have a huge number of overlays, how about taking the suggestion of @sofroniewn and just removing `CanvasOverlay` and `SceneOverlay` as distinct subclasses for now? They only add the `position` attribute in `CanvasOverlay`, which could just be added to the small number of canvas overlays we have right now.

I guess having distinction might introduce some nice clear semantics, but I also think that could come later.

Reply:
Ah right, I guess you explained that in: https://github.com/napari/napari/pull/4907/files#r946860221

I don't feel particularly strongly here.